src/Integration.UnitTests/Suppression/SonarQubeIssuesProviderTests.cs

```diff
@@ -335,37 +296,30 @@ public void GetIssues_IssuesNotYetFetch_WaitsForIssuesToBeFetched()
             var issue1 = new SonarQubeIssue("folder1/file1", "hash1", 0, "message", "sqkey:sqkey:projectID1",
                 SonarQubeIssueResolutionState.FalsePositive, "S101");
 
-            EventWaitHandle waitHandle = new EventWaitHandle(false, EventResetMode.ManualReset);
-
             int callbackCount = 0;
             bool callbackCompleted = false;
             Func<IList<SonarQubeIssue>> serviceFetchIssuesTask = () =>
             {
                 callbackCount++;
 
-                waitHandle.Set(); // signal so the test can continue
+                InitialFetchWaitHandle.Set(); // signal so the test can continue
 
                 Thread.Sleep(Debugger.IsAttached ? 5000 : 500);
                 callbackCompleted = true;
                 return new List<SonarQubeIssue> { issue1 };
             };
 
-            SetupSolutionBinding(isBound: true, isConnected: true, projectKey: "sqkey", issues: new List<SonarQubeIssue> { issue1 });
-            mockSqService.Setup(x => x.GetSuppressedIssuesAsync(It.IsAny<string>(), It.IsAny<CancellationToken>()))
-                .ReturnsAsync(serviceFetchIssuesTask)
-                .Verifiable();
+            SetupSolutionBinding(isConnected: true, serviceFetchIssuesTask: serviceFetchIssuesTask);
```

Comment:
Minor test refactoring: I added an overload for _SetupSolutionBinding_ so the tests can specify either the issues they want the service to return, or the delegate to execute.